docs/source/developers/contributing.rst

```diff
@@ -17,53 +17,73 @@
 
 .. _contributing:
 
-***********************
-Contribution Guidelines
-***********************
+****************************
+Contributing to Apache Arrow
+****************************
 
-There are many ways to contribute to Apache Arrow:
+Thanks for your interest in the Apache Arrow project. Arrow is a large project
+and may seem overwhelming when you're first getting involved.
+Contributing code is great, but that's probably not the first place to start.
+There are lots of ways to make valuable contributions to the project and
+community.
 
-* Contributing code (we call them "patches")
-* Writing documentation (another form of code, in a way)
-* Participating in discussions on `JIRA <https://issues.apache.org/jira/projects/ARROW/issues>`_ or the `mailing list <https://lists.apache.org/list.html?dev@arrow.apache.org>`_
-* Helping users of the libraries
-* Reporting bugs and asking questions
+This page provides some orientation for how to get involved. It also offers
+some recommendations on how to get best results when engaging with the
+community.
 
-Mailing List
-============
+Join the mailing lists
+======================
 
+A good first step to getting involved in the Arrow project is to join the
+mailing lists and participate in discussions where you can.
 Projects in The Apache Software Foundation ("the ASF") use public, archived
 mailing lists to create a public record of each project's development
-activities and decision-making process. As such, all contributors generally
-must be subscribed to the dev@arrow.apache.org mailing list to participate in
-the community.
+activities and decision-making process.
+While lacking the immediacy of chat or other forms of communication,
+the mailing lists give participants the opportunity to slow down and be
+thoughtful in their responses, and they help developers who are spread across
+many timezones to participate more equally.
 
-Note that you must be subscribed to the mailing list in order to post to it. To
-subscribe, send a blank email to dev-subscribe@arrow.apache.org.
+See `the community page <https://arrow.apache.org/community/>`_ for links to
+subscribe to the mailing lists and to view archives.
 
-Mailing list archives can be found `here <https://lists.apache.org/list.html?dev@arrow.apache.org>`_.
+Report bugs and request features
```

Comment:
```suggestion
Report bugs and propose features
```

(not necessarily better language, but just needed to think about how we sometimes complain about users gratuituously "requesting" features in the meaning of expecting them to be implemented)